dw-button.js

```diff
@@ -28,90 +28,113 @@ export class DwButton extends buttonFocus(Button) {
-          --mdc-theme-on-primary:  var(--mdc-theme-error);
+          --mdc-theme-primary: var(--mdc-theme-surface);
+          --mdc-theme-on-primary: var(--mdc-theme-error);
         }
-        :host([disabled]) .mdc-button:disabled  {
+        :host([disabled]) .mdc-button:disabled {
           color: var(--mdc-theme-text-disabled);
         }
 
         /** primary-text button  css start */
-        :host(.text) .mdc-button  {
+        :host(.text) .mdc-button {
           --mdc-theme-primary: var(--mdc-theme-text-primary);
         }
 
         /** outline button  css start */
-        :host([outlined]) .mdc-button--outlined:disabled  {
+        :host([outlined]) .mdc-button--outlined:disabled {
           --mdc-theme-primary: var(--mdc-theme-secondary);
         }
-        :host([error]) .mdc-button--outlined:disabled  {
+        :host([error]) .mdc-button--outlined:disabled {
           --mdc-theme-primary: var(--mdc-theme-error);
         }
-        :host([disabled][outlined]) .mdc-button--outlined:disabled  {
+        :host([disabled][outlined]) .mdc-button--outlined:disabled {
           border-color: var(--mdc-theme-text-disabled);
         }
 
         /** filled button start */
-        :host([filled]) .mdc-button:not(:disabled){
+        :host([filled]) .mdc-button:not(:disabled) {
           background-color: var(--mdc-theme-primary);
-          color: var(--mdc-theme-on-primary); 
+          color: var(--mdc-theme-on-primary);
         }
         :host([filled].secondary) .mdc-button:not(:disabled) {
           background-color: var(--mdc-theme-secondary);
-          color: var(--mdc-theme-on-secondary); 
+          color: var(--mdc-theme-on-secondary);
         }
         :host([filled][disabled]) .mdc-button:disabled {
-          background-color: var( --dw-disabled-background-color, var(--mdc-theme-divider-on-light));
+          background-color: var(--dw-disabled-background-color, var(--mdc-theme-divider-on-light));
         }
 
         .trailing-icon {
           display: flex;
           align-items: center;
         }
-        `
+      `,
     ];
   }
 
   static get properties() {
     return {
-      
       /**
        * When you want to filled this button set to `true`.
        */
-      filled : {
+      filled: {
         type: Boolean,
-        reflect: true
-      }
-    }
+        reflect: true,
+      },
+
+      /**
+       * whether to apply a focus effect on the button or not.
+       */
+      _noFocusEffect: { type: Boolean, reflect: true, attribute: "no-focus-effect" },
+    };
   }
 
-  constructor(){
+  constructor() {
     super();
     this.filled = false;
+    this._noFocusEffect = false;
   }
 
-  connectedCallback(){
+  connectedCallback() {
     super.connectedCallback();
-    /**
-    * call blur method to fix ripple effect after button click.
-    */
-    this.addEventListener('click', () => { this.blur(); })
+
+    this.addEventListener("mousedown", this._onMouseDown);
```

Comment:
As you have used `mousedown` it will work with mouse interface only. I assume, this fix won't work for the touch devices. Did you tested on touch device too?

I suggest to use `pointerdown` (https://developer.mozilla.org/en-US/docs/Web/API/Element/pointerdown_event) evnet for the generic implementation.